db-service/lib/deep-queries.js

```diff
@@ -114,6 +150,24 @@ const _calculateExpandColumns = (target, data, expandColumns = [], elementMap =
       expandColumns.push(expandColumn)
     }
 
+    // for deep update check if is recursiv and has looping association --> would result in cqn4sql loop
```

Comment:
The check does not actually do what the comment states it does. As recursive `compositions` can have any number of `compositions` in between. I don't see a `while(true)` or recursive function call in side this check which would be required to find the cycle.